consensus/metricsVoting.go

```diff
@@ -57,10 +61,37 @@ func (p *MetricsPeersVoting) UpdateCandidates() {
 	util.LogInfo(p.Log, util.CoWhite, fmt.Sprintf("@@UpdateCandidates:%d Candidates!", len(Candidates)))
 }
 
+//UpdateVotingMetrics Register first and get info for voting metrics. This is an  efficient way to get data without create a new stream
+func (p *MetricsPeersVoting) UpdateVotingMetrics(id peerlib.ID) error {
+	cctx, cancel := context.WithTimeout(context.Background(), waitingRespTime)
+	defer cancel()
+	s, err := p.watchNode.Host.NewStream(cctx, id, protocol.ID(p2p.TopicP2P))
```

Comment:
I am not sure about the purpose, since every time function called will create a new one from `NewStream`, do you actually want to save the variable `s` instead of creating new one everytime?

Reply:
create a new stream is ok. The idea is to reuse it is better, especially when it has many clients or in a resource-limited device

Reply:
I don't understand the meaning of `reuse`, since every time, stream & io are created new, does this way get reused?

```
	s, err := p.watchNode.Host.NewStream(cctx, id, protocol.ID(p2p.TopicP2P))
        rw := bufio.NewReadWriter(bufio.NewReader(s), bufio.NewWriter(s))
```

Reply:
I misunderstood your suggestion. It sounds a good suggestion. I did not re-use the code as I did before separating the package.